mos-platform/atari2600-common/vcslib.h

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 LLVM-MOS Project
+// Licensed under the Apache License, Version 2.0 with LLVM Exceptions.
+// See https://github.com/llvm-mos/llvm-mos-sdk/blob/main/LICENSE for license
+// information.
+
+#ifndef _VCSLIB_H
+#define _VCSLIB_H
+
+#include <atari2600.h>
+#include <stdint.h>
+
+#if defined(__ATARI2600_MAPPER__)
+#include <mapper.h>
+#endif
+
+// Define data types for different data sizes and signedness
+
+typedef unsigned char byte;   // 8-bit unsigned data type
+typedef signed char sbyte;    // 8-bit signed data type
+typedef unsigned short word;  // 16-bit unsigned data type
+
+// Atari 2600 kernel helpers, called in a sequence every frame
+
+__attribute__((leaf)) void kernel_1(void);	// before preframe
+__attribute__((leaf)) void kernel_2(void);	// before kernel
+__attribute__((leaf)) void kernel_3(void);	// after kernel
+__attribute__((leaf)) void kernel_4(void);	// after postframe
+
+// Function to set horizontal position of a game object.
+
+typedef enum { PLAYER_0=0, PLAYER_1, MISSILE_0, MISSILE_1, BALL } TIAObject;
+
+// A = X coordinate
+// X = object index
+__attribute__((leaf)) void _set_horiz_pos(byte xpos, byte objindex);
+// swap order of call
+#define set_horiz_pos(objindex,xpos) _set_horiz_pos(xpos,objindex);
```

Comment:
Instead, just make this a regular function. The compiler should almost always inline it, but it allows function pointers to work.

Reply:
Strangely enough, I tried inlining this function but it refused even with always_inline. It specialized the function since it was called with one of the parameters fixed, but it didn't inline either function.